codex-rs/core/src/command_safety/is_dangerous_command.rs

```diff
@@ -0,0 +1,104 @@
+use crate::bash::try_parse_bash;
+use crate::bash::try_parse_word_only_commands_sequence;
+
+const DANGEROUS_COMMANDS: &[&str] = &["git reset"];
+
+pub fn command_might_be_dangerous(command: &[String]) -> bool {
+    if contains_dangerous_command(command) {
+        return true;
+    }
+
+    // Support `bash -lc "<script>"` where the any part of the script might contain a dangerous command.
+    if let [bash, flag, script] = command
+        && bash == "bash"
+        && flag == "-lc"
+        && let Some(tree) = try_parse_bash(script)
+        && let Some(all_commands) = try_parse_word_only_commands_sequence(&tree, script)
```

Comment:
This seems to "fail deadly". If we aren't able to parse the script, should we presume it to be dangerous, rather than assuming it to be safe?